lib/cog_api/http/roles.ex

```diff
@@ -7,8 +7,8 @@ defmodule CogApi.HTTP.Roles do
     Base.get(endpoint, "roles") |> Base.format_response("roles", [%Role{}])
   end
 
-  def show(%Endpoint{}=endpoint, id) do
-    Base.get(endpoint, "roles/#{id}") |> Base.format_response("role", %Role{})
+  def show(%Endpoint{}=endpoint, role_name) do
+    Base.get_by(endpoint, "roles", name: role_name) |> Base.format_response("role", %Role{})
   end
```

Comment:
Is there a reason this is changing? It looks like `roles#show` in cog [already takes an `id`](https://github.com/operable/cog/blob/master/web/controllers/v1/role_controller.ex#L31).